File: 2001-number-of-pairs-of-interchangeable-rectangles/2001-number-of-pairs-of-interchangeable-rectangles.py
class Solution:
    def interchangeableRectangles(self, r: List[List[int]]) -> int:
        ans = 0
        dic = {}
        for i in range(len(r)):
            x = r[i][0]
            y = r[i][1]
            z = x/y
            if z in dic:
                ans+=dic[z]
                dic[z]+=1
            else:
                dic[z]=1
        return ans

File: 2001-number-of-pairs-of-interchangeable-rectangles/2001-number-of-pairs-of-interchangeable-rectangles.cpp
class Solution {
public:
    long long interchangeableRectangles(vector<vector<int>>& rectangles) {
        long long ans =0;
        unordered_map<double,int>ump;
        for(auto &x:rectangles){
            double a = x[0],b=x[1];
            // if(a>b) swap(a,b);
            double y = b/a;
            if(ump.find(y)!=ump.end()){
                ans+=ump[y];
            }
            ++ump[y];
        }
        return ans;
    }
};